### ISSUE TITLE
Unsigned integer codegen bug

### ISSUE THREAD

**Issue by markus-oberhumer** [nim-lang/Nim#6839]
This is a Nim codegen bug as can be seen by ` if (!NIM_TRUE) goto ...` in the generated C file.

Detected while cross-compiling using a 64-bit host and targeting a 32-bit machine.

```nim
when sizeof(int) == 4:
  var a = not 0'u
  doAssert a is uint
  doAssert a == 4294967295'u

  template foo(): uint = not 0'u

  var b = foo()
  doAssert b is uint
  doAssert b == 4294967295'u

  doAssert foo() is uint
  doAssert foo() == 4294967295'u    # FAILS !!!
```

**Comment by markus-oberhumer** [comment]
The problem is also easy to spot when targeting JavaScript using `nim js bug.nim`:

```
when sizeof(int) == 4:
  proc test() =
    let a = not 0'u
    doAssert a is uint
    doAssert a == 4294967295'u

    let b = uint(-1)
    doAssert b is uint
    doAssert b == 4294967295'u

  static: test() # does not compile
  test()         # asserts at runtime
```

**Comment by markus-oberhumer** [comment]
Just tested this and it seems the bug is fixed with nim-1.0.4.
